crates/next-core/src/next_import_map.rs

```diff
@@ -16,13 +18,18 @@ use crate::{
 
 /// Computes the Next-specific client import map.
 #[turbo_tasks::function]
-pub fn get_next_client_import_map(
+pub async fn get_next_client_import_map(
     project_path: FileSystemPathVc,
     ty: Value<ClientContextType>,
-) -> ImportMapVc {
+    next_config: NextConfigVc,
+) -> Result<ImportMapVc> {
     let mut import_map = ImportMap::empty();
 
-    insert_next_shared_aliases(&mut import_map, project_path);
+    insert_next_shared_aliases(
+        &mut import_map,
+        project_path,
+        &resolve_alias_options_to_alias_map(&*next_config.resolve_alias_options().await?).await?,
```

Comment:
This is a little awkward: `insert_next_shared_aliases` is a non-turbo, non-async function. Any suggestions that might clean this up?

Reply:
You should be able to call a non-turbo function with Vcs, it just needs to be async. The dependency tracking will be assigned to the closest turbo function in the call stack (this function in this case).